src/Viewer/components/Monaco/MonacoInstance.js

```diff
@@ -100,21 +95,17 @@ function MonacoInstance ({
 }) {
     const {theme} = useContext(ThemeContext);
     const editorRef = useRef(null);
-    const monacoRef = useRef(null);
+    const editorContainerRef = useRef(null);
     const timeoutRef = useRef(null);
-    const [monacoTheme, setMonacoTheme] = useState("customLogLanguageLight");
-
-    const [language, setLanguage] = useState("");
-
-    loader.config({monaco});
+    const getMonacoThemeName = (theme) => (
+        (theme === THEME_STATES.LIGHT) ? "customLogLanguageLight" : "customLogLanguageDark"
+    );
 
     /**
      * Called before the monaco editor is mounted.
-     *
-     * @param {object} monaco
      */
-    function handleEditorWillMount (monaco) {
-        beforeMount(monaco);
+    const _beforeMount = () => {
```

Comment:
Any particular reason for the names prefixed with `_`? We used to use the `_` to indicate a private method in a class. Of course, nowadays you have truly [private methods](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_properties).

Reply:
If it's just to avoid the name conflict with the passed-in callback, how about renaming the passed in callbacks with the suffix `Callback` (e.g., `beforeMountCallback`)?